src/event/event_windows.c

```diff
@@ -364,15 +364,19 @@ _dispatch_socket_callback(PTP_CALLBACK_INSTANCE inst, void *context,
 		} else if (lNetworkEvents & FD_READ) {
 			ioctlsocket(sock, FIONREAD, &dwBytesAvailable);
 		}
-		if (lNetworkEvents & FD_READ) {
+		// ioctlsocket could legally return 0 and that doesn't mean the socket is closed.
+		// Here and below we have to double-check dwBytesAvailable and make sure we don't 
+		// wake up read/write handlers with 0 value for live (not closed) socket, because
+		// that would force handler to shut down.
+		if (((lNetworkEvents & FD_READ) && dwBytesAvailable > 0) || (lNetworkEvents & FD_CLOSE)) {
```

Comment:
This seems plausible to occur.  But `FD_READ` is already set when `FD_CLOSE` is set, so this is rather confusing.